### PR TITLE
Fix missing link to required library in Windows when XLINK_LIBUSB_SYSTEM is ON

### PR DESCRIPTION
**Pull request by traversaro**
The `PathCchRemoveFileSpec` function contained in `Pathcch.lib` is called in the code in Windows (see https://github.com/luxonis/XLink/blob/6d2a0dd2e7f0f08657b492ec684a493d7ab1c2a0/src/pc/Win/src/win_usb_host.cpp#L39), so the `Pathcch.lib` library should be linked even if  XLINK_LIBUSB_SYSTEM is set to ON.

### CODE REVIEW

**Comment by traversaro** [comment]
The Windows failure is unrelated and related to https://github.com/luxonis/depthai-core/pull/1037 and https://github.com/cpp-pm/hunter/pull/747 .

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,12 +75,16 @@ else()
...
     if(WIN32 AND NOT MINGW)
-        target_link_libraries(${TARGET_NAME} PRIVATE delayimp.lib Pathcch.lib)
+        target_link_libraries(${TARGET_NAME} PRIVATE delayimp.lib)
         # workaround https://gitlab.kitware.com/cmake/cmake/-/issues/20022
         target_link_options(${TARGET_NAME} PUBLIC "$<LINK_ONLY:/DELAYLOAD:libusb-1.0$<$<CONFIG:Debug>:d>.dll>")
     endif()
 endif()
 
+if(WIN32 AND NOT MINGW)
+    target_link_libraries(${TARGET_NAME} Pathcch.lib)
```
Comment (themarpe) [inline CMakeLists.txt:85]:
```suggestion
    target_link_libraries(${TARGET_NAME} PRIVATE Pathcch.lib)
```

Shouldn't this remain `PRIVATE` as before? (not sure if that is the default)

**Review by themarpe — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (traversaro) [reply CMakeLists.txt]: Sure, nice catch! `PRIVATE` is the default but you can't mix plain and keyword signatures (see https://cmake.org/cmake/help/latest/policy/CMP0023.html), so adding `PRIVATE` is totally the correct thing to do.

**Comment by traversaro** [comment]
The merge confused the PR, let me do it from scratch.

**Comment by traversaro** [comment]
> The merge confused the PR, let me do it from scratch.

Ok, now it is clean, let me know if it is ok to merge with squash or you prefer that I cleanup the history.

**Pull request merged**